sirepo/template/flash.py

```diff
@@ -683,6 +666,14 @@ def _generate_parameters_file(data):
     return res
 
 
+def _grid_evolution_columns(run_dir):
+    try:
+        with open(run_dir.join(_GRID_EVOLUTION_FILE)) as f:
```

Comment:
we shouldn't have naked opens. i think this has to be pkcompat

Reply:
We have naked opens all over the template code. Is there something different about this one or are they all wrong?

In pkcompat what would the open function do? Require that we always specify an encoding so Python doesn't default?

Reply:
Actually, we added pkio.open_text for this reason. The only places where there are naked "text" opens is where I haven't gotten to fixing them yet, e.g. ml.py. Most of the opens are binary. The proper form is e.g. from srw.py:

```txt
with pkio.open_text(filename) as f:
```

I would like avoid proliferating decoding of utf-8 everywhere.

Reply:
These file are [ASCII encoded](http://flash.uchicago.edu/~jbgallag/2012/flash4_ug/node15.html#SECTION05224000000000000000). Should I still use pkio.open_text? Will we ever run on a machine local.getprefferedencoding() isn't ascii compatible?

Reply:
@robnagler ping

Reply:
I'm getting concerned about getprefferedencoding. I think we might want to use utf-8 instead, because fundamentally, pkio.open_text is not how we would handle i18n. It would be explicit based on the logged in user (not command line user).

Therefore, I think the change is to open_text always be utf8. We have to leave pkcompat.locale_str the way it is, because locale means something. We might want to change those references anyway to something else eg. to_bytes and from_bytes.

Reply:
https://github.com/radiasoft/sirepo/pull/2676/commits/4dfa1f80bac0b73eb40a7d8189d81a158474af16